janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -196,11 +199,15 @@ public void createMapping(String indexName, String typeName, Map<String,Object>
     }
 
     @Override
-    public Map getMapping(String indexName, String typeName) throws IOException{
-        Response response = performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR + indexName + "/_mapping/" + typeName, null);
+    public Pair<Boolean, Map> getMapping(String indexName, String typeName) throws IOException{
+        final Response response = performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR + indexName + "/_mapping/" + typeName, null);
         try (final InputStream inputStream = response.getEntity().getContent()) {
-            Map<String, RestIndexMappings> settings = mapper.readValue(inputStream, new TypeReference<Map<String, RestIndexMappings>>() {});
-            return settings == null ? null : settings.get(indexName).getMappings().get(typeName).getProperties();
+            final Map<String, RestIndexMappings> settings = mapper.readValue(inputStream, new TypeReference<Map<String, RestIndexMappings>>() {});
+            if (settings == null) {
+                return null;
+            }
+            final RestIndexMapping mappings = settings.get(indexName).getMappings().get(typeName);
```

Comment:
What about changing the interface to return the index mapping instead of Pair object? So you'd pull out `RestIndexMapping` out of the `rest` package and call it `ElasticSearchIndexMapping` or something. Or is it too ugly?